components/emoji_picker/emoji_picker_overlay.jsx

```diff
@@ -59,6 +59,16 @@ export default class EmojiPickerOverlay extends React.PureComponent {
     }
 
     render() {
+        const emojiTrigger = this.props.target();
+        let rightOffset = Constants.DEFAULT_EMOJI_PICKER_RIGHT_OFFSET;
+        if (emojiTrigger) {
+            rightOffset = window.innerWidth - emojiTrigger.getBoundingClientRect().left - Constants.DEFAULT_EMOJI_PICKER_LEFT_OFFSET;
```

Comment:
Using the force, I'm worried about performance and doing this on `render`. Is there a solution that lets us compute this when the component is mounted instead (potentially leaving it at the wrong place if the window is resized)?

Reply:
Not sure, up for suggestions.
I was just following what we already do in:
https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/reaction_list/reaction_list.jsx

Reply:
There is only one prop here which should be changing i.e show or hide but we could do this on mount and change position based on a resize listener to avoid potential perf regression if we add new props.